clusterman/migration/worker.py

```diff
@@ -69,7 +83,17 @@ def _monitor_pool_health(manager: PoolManager, timeout: float, drained: Collecti
         draining_happened = draining_happened or not any(
             node.agent.agent_id == connector.get_agent_metadata(node.instance.ip_address).agent_id for node in drained
         )
-        if draining_happened and manager.is_capacity_satisfied() and not connector.get_unschedulable_pods():
+        if (
+            draining_happened
+            and manager.is_capacity_satisfied()
+            and not (
+                check_pods
+                and any(
+                    reason == PodUnschedulableReason.InsufficientResources
+                    for _, reason in connector.get_unschedulable_pods()
+                )
+            )
```

Comment:
thoughts on extracting this to a variable (or function if we wanna keep the short-circuiting behavior)? 10 lines for an if-statement is getting pretty long (+ the multiple levels of indentation) and naming the `any()` check will help this be more self-documenting